### PR TITLE
fixing some issues

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qiita_db/util.py [hunk 1]
```diff
@@ -1486,17 +1488,16 @@ def generate_study_list(user, visibility):
 
                 # cleaning aids_with_deprecation
                 info['artifact_biom_ids'] = []
-                info['preparation_data_types'] = []
                 if info['aids_with_deprecation'] is not None:
                     for x in info['aids_with_deprecation']:
-                        # f1-2 are the default names given by pgsql
+                        # f1-3 are the default names given by pgsql
```
Comment (charles-cowart) [inline qiita_db/util.py]: Just to confirm, f1-3 I assume means (f1, f2, f3). You've increased the range by 1, but the conditional below is still testing only for not x['f2']. This is still okay, right?

File: qiita_pet/handlers/study_handlers/artifact.py [hunk 2]
```diff
@@ -51,9 +51,16 @@ def post(self):
...
         # Request the rest of the arguments, which will be the files
-        files = {arg: self.get_argument(arg) for arg in self.request.arguments
-                 if arg not in ['name', 'prep-template-id', 'artifact-type',
-                                'import-artifact']}
+        files = dict()
+        for arg in self.request.arguments:
+            if arg not in ['name', 'prep-template-id', 'artifact-type',
+                           'import-artifact']:
+                arg_name = arg
+                # removing ending [], in case they exists, necessary for JS
```
Comment (charles-cowart) [inline qiita_pet/handlers/study_handlers/artifact.py]: I believe exists -> exist

File: qiita_db/artifact.py [hunk 3]
```diff
@@ -740,7 +740,7 @@ def visibility(self, value):
 
             # then let's check that the sample/prep info files have the correct
             # restrictions
-            if value != 'sandboxed' and study is not None:
+            if value != 'sandbox' and study is not None:
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/artifact.py]: Yup, there is an open issue: https://github.com/biocore/qiita/issues/410

File: qiita_db/util.py (hunk 1 above)
Comment (antgonza) [reply qiita_db/util.py]: Yeah, good catch. Originally we had 3 but the comment only mentioned 2, then changed the code to only use 2 and forgot to bring the comment back ... changing ...

**Pull request merged**